osu.Game/Database/RealmAccess.cs

```diff
@@ -966,6 +967,27 @@ private void applyMigrationsForVersion(Migration migration, ulong targetVersion)
 
                     break;
                 }
+
+                case 31:
+                {
+                    var scores = migration.NewRealm
+                                          .All<ScoreInfo>()
+                                          .Where(s => !s.IsLegacyScore)
+                                          .Where(s => s.RulesetID == 0);
```

Comment:
```suggestion
                                          .Filter(@"IsLegacyScore == false && Ruleset.OnlineID == 0");
```

Must be done using the realm query language, else the migration fails and corrupts db because realm doesn't support reading non-persisted properties (`s.RulesetID` is a wrapper over a `s.RulesetInfo.OnlineID` access).